### PR TITLE
[web][api][formio][workflow]Approve Application Exceptions - Part 2

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- **Code smells due to the ES5 javascript file included for the workflow.**
- Form.io definitions are still following the old format to allow the proper diff. I will change after this PR.
- Added a new verification for the assessment-gateway workflow to check for exceptions (already deployed to DEV).
image

image

- Student Application forms were changed to add hidden fields to identify the Student Application Exceptions, as shown below.

image

- Created the form to approve, decline, and view the application exceptions.

**Approved**

image

**Declined**

image

**Pending**

image

- New components and layout adjustments as shown below.

image

image

### CODE REVIEW

File: sources/packages/api/src/services/workflow/workflow-actions.service.ts [hunk 1]
```diff
@@ -159,6 +159,37 @@ export class WorkflowActionsService {
...
+            type: "string",
+          },
+        },
+      });
+    } catch (error: unknown) {
+      this.logger.error(
+        `Error while sending application exception approval message applicationExceptionId: ${applicationExceptionId}.`,
+      );
+      this.logger.error(error);
+      // The error is not thrown here, as we are failing silently.
```
Comment (henryoforeh-aot) [inline sources/packages/api/src/services/workflow/workflow-actions.service.ts:189]: Please move this comment to somewhere between line 184 and 185.

File: sources/packages/web/src/components/aest/students/assessment/Request.vue [hunk 2]
```diff
@@ -10,59 +10,71 @@
...
-              field="submittedDate"
-              header="Submitted date"
-              sortable="true"
-              ><template #body="slotProps">{{
-                dateOnlyLongString(slotProps.data.submittedDate)
-              }}</template></Column
+          <toggle-content :toggled="!requestedAssessment.length">
+            <DataTable
+              :value="requestedAssessment"
+              :paginator="true"
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/components/aest/students/assessment/Request.vue:16]: Is "true" a variable that is defined somewhere like the other ones (:rows="DEFAULT_PAGE_LIMIT", :rowsPerPageOptions="PAGINATION_LIST", etc) ?

File: sources/packages/web/src/services/ApplicationExceptionService.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  /**
+   * Get a student application exception detected after the student application was
+   * submitted, for instance, when there are documents to be reviewed.
+   * @param exceptionId exception to be retrieved.
+   * @returns student application exception information.
+   */
+  async getExceptionById(
+    exceptionId: number,
+  ): Promise<ApplicationExceptionAPIOutDTO> {
+    return ApiClient.ApplicationExceptionApi.getExceptionById(exceptionId);
```
Comment (henryoforeh-aot) [inline sources/packages/web/src/services/ApplicationExceptionService.ts:24]: Please use "await" here.

Comment (henryoforeh-aot) [reply sources/packages/web/src/services/ApplicationExceptionService.ts:24]: Please disregard my comment here.

File: sources/packages/api/src/services/workflow/workflow.models.ts [hunk 4]
```diff
@@ -8,6 +8,21 @@ export interface WorkflowStartResult {
...
+/**
+ * Variable value to be assigned to a message payload.
+ */
+export interface ProcessVariableValue {
+  value: string;
+  type: "string" | "integer";
+}
+
+/**
+ * variable to be send with a message payload.
```
Comment (ann-aot) [inline sources/packages/api/src/services/workflow/workflow.models.ts:20]: capital `V`

File: sources/packages/web/src/services/http/ApplicationExceptionApi.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import HttpBaseClient from "@/services/http/common/HttpBaseClient";
+import {
+  ApplicationExceptionAPIOutDTO,
+  UpdateApplicationExceptionAPIInDTO,
+} from "@/services/http/dto";
+
+/**
+ * Http API client for Student Application Exceptions, for instance, when a
```
Comment (ann-aot) [inline sources/packages/web/src/services/http/ApplicationExceptionApi.ts:8]: `Student Application Exceptions` lower case

File: sources/packages/web/src/views/aest/ApplicationExceptionsApproval.vue [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+/**
+ * Model to be used to populate the form.io.
+ */
+type ApplicationExceptionFormModel = Omit<
+  ApplicationExceptionAPIOutDTO,
+  "assessedDate"
+> & {
+  /**
+   * Exception status at the moment that the data was loaded.
+   * used manly when the form is being edited and change the
```
Comment (ann-aot) [inline sources/packages/web/src/views/aest/ApplicationExceptionsApproval.vue:51]: mainly?

**Review by ann-aot — COMMENTED** [review]
added some minor comments

File: sources/packages/api/src/services/workflow/workflow-actions.service.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/services/workflow/workflow-actions.service.ts:189]: Any specific reason? It is following exactly what the other 4 similar methods in the same file are doing.
The comment is placed exactly where the exception would be thrown. With a proper reason, we can change it.

File: sources/packages/web/src/components/aest/students/assessment/Request.vue (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/aest/students/assessment/Request.vue:16]: The change is not actually from this PR but what was the question here? Sorry, I am not following?

Comment (andrewsignori-aot) [reply sources/packages/web/src/components/aest/students/assessment/Request.vue:16]: Is it about using `:paginator="true"` vs `paginator="true"`?
If this is the case, the first one is a boolean value assigned to a boolean property, the second one is a string value assigned to a boolean property, that is my understanding.

**Review by ann-aot — APPROVED** [review]
Good Job With exception and components 👍 Thanks for doing the changes

**Review by henryoforeh-aot — APPROVED** [review]

File: sources/packages/api/src/services/workflow/workflow-actions.service.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/services/workflow/workflow-actions.service.ts:189]: Yes, you are right, but maybe this is not the case. The comment is not about the implementation above it. The comment is there to explain why there is a try/catch that is capturing the exception and them doing nothing. Line 189 was supposed to have something like `throw error` that is missing in these methods. I am not sure it makes better sense now.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrewsignori-aot

**Pull request merged**